app/models/audit/eligibility/EligibilityCheckAuditDetail.scala

```diff
@@ -74,8 +74,9 @@ object EligibilityCheckAuditDetail {
               } else {
                 (currentFieldName, transformedJsArray)
               }
-            case value if currentFieldName === fieldName => (currentFieldName, transformation(value))
-            case value if currentFieldName =!= fieldName => (currentFieldName, value)
+            case value if currentFieldName === fieldName                         => (currentFieldName, transformation(value))
+            case value if currentFieldName =!= fieldName                         => (currentFieldName, value)
+            case value @ (JsNull | JsNumber(_) | JsString(_) | JsTrue | JsFalse) => (currentFieldName, value)
```

Comment:
Compiler made me do it

Reply:
does something like this work to make it shorter?
```scala
case value if currentFieldName === fieldName => (currentFieldName, transformation(value))
case value: JsValue => (currentFieldname, value)
```

Reply:
I'll try it, compiler is so damn strict now (which is a good thing 😂 )

Reply:
Yeah it did, pushing now